### DISCUSSION TITLE
shared scripts between client and server

### DISCUSSION THREAD

**Discussion by victj99** [quasarframework/quasar#9298] [CLI - SSR mode] [answered]
Hi, I'm new to quasar, I'm experimenting with creating an app with rest api using ssr (you know if that's a good idea?) And typescript.
Well the issue is that if I want to have scripts with functions, classes, enums, etc and use them both on the client and server side
In which folder do I put them? In src-ssr? The root of the project? Or is there a specific folder? I have searched the documentation and the internet but I did not find anything
Thank you

**Comment by ontwikkelfabriek** [comment]
The ssr server Quasar uses is not for hosting/creating a API endpoints. It's just for server side rendering. For an api rest server you need to use something like Express. Quasar is frontend only.

A way to codeshare between Express(or any nodejs based server) and Quasar you could put the Express project folder inside of Quasar's project folder, for example `my_quasar_project_folder/api'. That way Quasar can import js files from you server in the 'api' folder.

**Comment by hawkeye64** [comment, marked as answer]
I've found the best way to share code like this, is to create a yarn workspace and set up 3 folders: client, backend, and shared.

**Reply by victj99** [reply]
yes, I was putting the endpoints inside "src-ssr / middlewares", as they answered before the ssr mode is not for that.
After hours of researching at the end I chose to use a monorepo with yarn workspaces (I had no idea about monorepos) where is my quasar app together with nestjs
I hope this helps someone else
PD google translate

**Comment by TobyMosque** [comment]
if u wanna u backend to be served from the same process of your frontend, u can follow that strategy.
https://dev.to/quasar-pt-br/monorepo-quasar-v2-nestjs-30jl

the advantage here, u can decouple the backend from the frontend at anytime.
in order to share code between the frontend and the backend, u just need to add a new TS proj to your workspace:
anyway, u'll need to configure the TS by yourself.
https://www.digitalocean.com/community/tutorials/typescript-new-project

**Comment by rstoenescu** [comment]
In Quasar v2 you can use a SSR Middleware to hook your API too. https://next.quasar.dev/quasar-cli/developing-ssr/ssr-middleware#anatomy-of-a-middleware-file
